async-query-core/src/test/java/org/opensearch/sql/spark/utils/SQLQueryUtilsTest.java

```diff
@@ -404,15 +407,40 @@ void testAutoRefresh() {
 
   @Test
   void testValidateSparkSqlQuery_ValidQuery() {
-    String validQuery = "SELECT * FROM users WHERE age > 18";
-    List<String> errors = SQLQueryUtils.validateSparkSqlQuery(validQuery);
+    DataSource dataSource = Mockito.mock(DataSource.class);
+    Mockito.when(dataSource.getConnectorType()).thenReturn(DataSourceType.PROMETHEUS);
+    String validQuery = "DELETE FROM Customers WHERE CustomerName='Alfreds Futterkiste'";
+    List<String> errors = SQLQueryUtils.validateSparkSqlQuery(dataSource, validQuery);
     assertTrue(errors.isEmpty(), "Valid query should not produce any errors");
   }
 
+  @Test
+  void testValidateSparkSqlQuery_SelectQuery_DataSourceSecurityLake() {
```

Comment:
I think we want to have more test cases for positive and negative cases, such as `SELECT` query is included as part of the `CREATE` query, etc. To avoid redundancy, you'd want to extract methods like `verifyValidSecurityLakeQuery`.

Reply:
added

Reply:
As I mentioned in the above comment, can we extract method and reduce redundancy?

Reply:
done